### PR TITLE
Webpack 2 support

### PR DESCRIPTION
**Pull request by renarsvilnis**
Initial pull request for #38 

Stuff that has been done:
- Add 'src/helpers.js' for commonly used functions
- Add JSDOC documentation for various functions
- Add support for Webpack v2
- Add enhanced support for Webpack v1 by reading 'resolve.root', 'resolve.fallback'

> Pull request needs feedback and is unfinished
> ⚠️  **Don't merge, yet!**

I'm worried that the enhanced Webpack v1 and Webpack v2 support might not work as they hold absolute paths (e.g. `path.join(__dirname, 'app')`) which in the `src/helpers.js@lookupLocal` function might not resolve correctly.

I'd love to see some help on adding documentation on remaining functions and especially `lookupLocal` would help me understand the library and finish the pull request.

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+'use babel';
```
Comment (nkt) [inline src/helpers.js]: Please create `utils/` folder and place every helper separately.

File: src/webpack-lookup.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+'use babel';
```
Comment (nkt) [inline src/webpack-lookup.js]: Move this to `utils/`

**Comment by renarsvilnis** [comment]
### Update

Did a overall file structure refactor:
- renamed `src` folder to `lib` (*common npm practice*)
- added top-level `index.js` that just exports the plugin instance (*common npm practice*)
- moved helper/private functions from `completion-provider.js` to `utils`.

I can also try adding testing and continuous-integration using `mocha`,  `chai`, `istanbul`, `travis`.

Should I close this branch and make a pull request to `v2` branch?

**Comment by nkt** [comment]
Atom has own test infrastructure. [Atom Linter](https://github.com/steelbrain/linter) as example of package with tests.
I prefer CircleCI, but travis is ok.

 > Should I close this branch and make a pull request to v2 branch?

No, I'll update master.

**Comment by nkt** [comment]
@renarsvilnis please add `circle.yml`:
```yaml
machine:
  node:
    version: 6
```

**Comment by renarsvilnis** [comment]
Added `circle.yml`.

**Comment by nkt** [comment]
@renarsvilnis is it still WIP?

**Comment by renarsvilnis** [comment]
Yes, haven't had time to figure out how `lookupLocal` function works and if it would do mad-stuff when passed absolute paths from the webpack config.

**Comment by nkt** [comment]
@renarsvilnis I guess `lookupLocal`, actually works not fine right now.
The algorithm should be:
 - split directory and filename ('/foo' and 'ba'), ('' and 'fo')
 - concat current file directory and split directory from prev step
 - lookup names in dirname from prev step
 - filter names using filename prefix from first step

But first step works not fine. For example, if prefix is `'..`, `'./'`, `'../..'` it's becomes `''`. So if you have better implementation, I'll glad to see it.

**Comment by nkt** [comment]
@renarsvilnis are you still interested in this changes?

**Comment by renarsvilnis** [comment]
@nkt Yeah, but I quite busy right now, can get back to it in a month.

**Comment by MatthieuLemoine** [comment]
@renarsvilnis are you still willing to work on this or should I give it a try ?

**Comment by renarsvilnis** [comment]
@MatthieuLemoine no, have forgotten about it. You can continue on it.

**Comment by jonyeezs** [comment]
Can this be close, @nkt @renarsvilnis ? @MatthieuLemoine seems to have solved it on PR #64 ?

**Comment by jonyeezs** [comment]
Superseded by #64

**Pull request closed without merging**